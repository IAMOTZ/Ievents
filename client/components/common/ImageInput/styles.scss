.react-dropzone,
.image-preview img,
.image-preview-overlay {
  height: 250px;
  width: 351px;
  border-radius: 3px;
}

.image-preview img, .react-dropzone {
  position: relative;
}

.image-preview-overlay {
  display: none;
  position: absolute;
  top: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.6);
  color: white;
  transition: all ease 2.0s;
  text-align: center;
  padding: 100px 0px;
  font-size: 16px;
}

.react-dropzone:hover, 
.image-preview:hover, 
.image-preview img:hover {
  cursor: pointer;
  .image-preview-overlay {
    display: block;
  }
}
