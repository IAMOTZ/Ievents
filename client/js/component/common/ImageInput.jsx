import React from 'react';
import Dropzone from 'react-dropzone';

const ImageInput = (props) => {
  if (props.newImage) {
    // Note that the new image is an image file
    return (
      <Dropzone
        multiple={false}
        accept="image/jpeg"
        onDrop={props.onDrop}>
        <a href={props.newImage.preview} target="blank">{props.newImage.name}</a>
      </Dropzone>
    )
  } else if (props.previousImage) {
    // Note that the previous image is a url 
    // [later] The previous image should also be displayed.
    return (
      <Dropzone
        multiple={false}
        accept="image/jpeg"
        onDrop={props.onDrop}>
        <div>
          <p>Drop a new image or click to select a new image to upload[.jpeg only] </p>
        </div>
      </Dropzone>
    );
  } else {
    return (
      <Dropzone
        multiple={false}
        accept="image/jpeg"
        onDrop={props.onDrop}>
        <p>Drop an image or click to select an image to upload[.jpeg only]</p>
      </Dropzone>
    );
  }
};

export default ImageInput;
