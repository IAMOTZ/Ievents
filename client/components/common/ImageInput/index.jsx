import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const ImageInput = (props) => {
  const { newImage, previousImage } = props;
  let captionText;
  let imageSource;
  if (newImage) {
    imageSource = newImage;
    captionText = 'Click to select another image.';
  } else if (previousImage) {
    imageSource = previousImage;
    captionText = 'Click to select another image.';
  } else {
    imageSource = defaultImage;
    captionText = 'This is the default center image. Click to select another image.';
  }
  return (
    <Dropzone
      className="react-dropzone"
      multiple={false}
      accept=".jpg, .jpeg, .png"
      onDropAccepted={props.onDrop}
    >
      <div className="image-preview">
        <img src={imageSource} alt="" />
        <div className="image-preview-overlay">
          <span>{captionText}</span>
        </div>
      </div>
    </Dropzone>
  );
};

ImageInput.defaultProps = {
  newImage: null,
  previousImage: null,
};

ImageInput.propTypes = {
  onDrop: PropTypes.func.isRequired,
  newImage: PropTypes.string,
  previousImage: PropTypes.string,
};

export default ImageInput;
