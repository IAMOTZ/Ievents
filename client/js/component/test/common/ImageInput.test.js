import React from 'react';
import ImageInput from '../../common/ImageInput.jsx';

const props = {
  previousImage: null,
  newImage: null,
  style: {},
  onDrop() {},
};

const alterProps = newProps => ({ ...props, ...newProps });

describe('<ImageInput />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ImageInput />);
  });
  it('should render a link if there is a new image', () => {
    const newImage = {
      name: 'Test',
      preview: 'http://jsutATestImage.png',
    };
    const expectedLink = <a href={newImage.preview} target="blank">{newImage.name}</a>;
    wrapper.setProps(alterProps({ newImage }));
    expect(wrapper.contains(expectedLink)).toBeTruthy();
  });
  it('should render text for adding a new image if there was a previous image', () => {
    const previousImage = {
      name: 'Test',
      preview: 'http://jsutATestImage.png',
    };
    const expectedText = <p>Drop a new image or click to select a new image to upload[.jpeg only]</p>;
    wrapper.setProps(alterProps({ previousImage }));
    expect(wrapper.contains(expectedText)).toBeTruthy();
  });
  it('should render text for adding an image if there was neither previous or new image', () => {
    const expectedText = <p>Drop an image or click to select an image to upload[.jpeg only]</p>;
    expect(wrapper.contains(expectedText)).toBeTruthy();
  });
});
