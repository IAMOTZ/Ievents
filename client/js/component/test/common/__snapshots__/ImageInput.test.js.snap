// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageInput /> should render a link if there is a new image 1`] = `
<CustomDropZone
  onDrop={[Function]}
  style={Object {}}
>
  <a
    href="http://jsutATestImage.png"
    target="blank"
  >
    Test
  </a>
</CustomDropZone>
`;

exports[`<ImageInput /> should render text for adding a new image if there was a previous image 1`] = `
<CustomDropZone
  onDrop={[Function]}
  style={Object {}}
>
  <p>
    Drop a new image or click to select a new image to upload[.jpeg only]
  </p>
</CustomDropZone>
`;

exports[`<ImageInput /> should render text for adding an image if there was neither previous or new image 1`] = `
<CustomDropZone>
  <p>
    Drop an image or click to select an image to upload[.jpeg only]
  </p>
</CustomDropZone>
`;
